filebeat/input/filestream/filestream.go

```diff
@@ -297,3 +305,11 @@ func (f *logFile) Close() error {
 	f.log.Debugf("Closed reader. Path='%s'", f.file.Name())
 	return err
 }
+
+// updateOffset updates the offset and lastTimeRead atomically
+func (f *logFile) updateOffset(delta int) {
+	f.offsetMutx.Lock()
+	f.offset += int64(delta)
+	f.lastTimeRead = time.Now()
+	f.offsetMutx.Unlock()
+}
```

Comment:
This looks like it might potentially affect the performance. Do you think we should run benchmarks to confirm no impact?

Reply:
I meant a benchbuilder run against this branch. This function is quite a hot path.